embedchain/vectordb/zilliz.py

```diff
@@ -0,0 +1,206 @@
+from typing import Dict, List, Optional
+
+from embedchain.config import ZillizDBConfig
+from embedchain.helper.json_serializable import register_deserializable
+from embedchain.vectordb.base import BaseVectorDB
+
+try:
+    import pymilvus
+    from pymilvus import MilvusClient
+    from pymilvus import connections, FieldSchema, CollectionSchema, DataType, Collection, utility
+except ImportError:
+    raise ImportError(
+        "Zilliz requires extra dependencies. Install with `pip install --upgrade embedchain[milvus]`"
+    ) from None
+
+
+@register_deserializable
+class ZillizVectorDB(BaseVectorDB):
+    """Base class for vector database."""
+
+    def __init__(self, config: ZillizDBConfig = None):
+        """Initialize the database. Save the config and client as an attribute.
+
+        :param config: Database configuration class instance.
+        :type config: ZillizDBConfig
+        """
+
+        if config is None:
+            self.config = ZillizDBConfig()
+        else:
+            self.config = config
+
+        try:
+            self.client = MilvusClient(
+                uri=self.config.uri,
+                token=self.config.token,
+            )
+
+            self.connection = connections.connect(
+                uri=self.config.uri,
+                token=self.config.token,
+            )
+        except Exception:
+            raise ValueError("Credentials Provided are not Valid")
+
+        super().__init__(config=self.config)
+
+    def _initialize(self):
+        """
+        This method is needed because `embedder` attribute needs to be set externally before it can be initialized.
+
+        So it's can't be done in __init__ in one step.
+        """
+        self._get_or_create_collection(self.config.collection_name)
+
+    def _get_or_create_db(self):
+        """Get or create the database."""
+        return self.client
+
+    def _get_or_create_collection(self, name):
+        """
+        Get or create a named collection.
+
+        :param name: Name of the collection
+        :type name: str
+        """
+        if utility.has_collection(name):
+            self.collection = Collection(name)
+        else:
+            fields = [
+                FieldSchema(name="id", dtype=DataType.VARCHAR, is_primary=True, max_length=512),
+                FieldSchema(name="doc", dtype=DataType.VARCHAR, max_length=2048),
```

Comment:
Do we have to define max_length here? Also, can we change the name of the field to `text` for consistency?